File: backendSharp/Models/Position.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Trowoo.ViewModels;

namespace Trowoo.Models
{
    public class Position
    {
        public int Id {get; set;}
        [Required]
        public DateTime OpenedDate {get; set;}
        [Required]
        public decimal Cost {get; set;}
        public Security Security {get; set;}
        public TrailingStop TrailingStop {get; set;}
        public LowPrice LowPrice {get; set;}
        public HighPrice HighPrice {get; set;}
        public PositionPerformanceMetrics PositionPerformanceMetrics {get; set;}
    }
}

File: client/src/app/shared/models/trowoo/position.ts
import { Security } from './security';
import { TrailingStop } from './trailingStop';
import { LowPrice } from './lowPrice';
import { HighPrice } from './highPrice';
import { PositionPerformanceMetrics } from './positionPerformanceMetrics';

export interface Position {
  id: number;
  openedDate: Date;
  cost: number;
  security: Security;
  trailingStop?: TrailingStop;
  lowPrice?: LowPrice;
  highPrice?: HighPrice;
  positionPerformanceMetrics?: PositionPerformanceMetrics;
}
